### PR TITLE
Bounding-box collision detection

### PR DESCRIPTION
**Pull request by wbrbr**

### CODE REVIEW

File: src/collision.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include "collision.h"
+#include "game.h"
+
+bool rect_collide_x(SDL_Rect rect1, SDL_Rect rect2) {
+    return false;
+}
+
+bool rect_collide_y(SDL_Rect rect1, SDL_Rect rect2) {
+    return false;
+}
```
Comment (vhakulinen) [inline src/collision.c]: These could be removed?

**Comment by wbrbr** [comment]
Done :+1:

File: src/entity.c [hunk 2]
```diff
@@ -10,6 +11,8 @@ entity* CreateEntity(SDL_Renderer *ren, int x, int y, char *imagePath) {
     entity *e = malloc(sizeof(entity));
     e->x = x;
     e->y = y;
+    e->w = w;
+    e->h = h;
```
Comment (vhakulinen) [inline src/entity.c:15]: Saving width and height to entity properties means that in future, when changing the sprite, we'll need to remember to change `w` and `h` properties. Is this better than just querying the texture on collision detection function?

File: src/game.c [hunk 3]
```diff
@@ -14,10 +14,11 @@ void game_init(SDL_Renderer *ren) {
...
 	char* path = buildPath(ASSETS, "sprites/stickman.bmp");
-    entities[0] = CreateEntity(ren, 200, 200, path);
-    entities[0]->velocity = 30;
 
-    character = CreateEntity(ren, 0, 0, path);
+    character = CreateEntity(ren, 100, 100, 64,128,path);
     character->type = ENTITY_TYPE_MAIN_CHARACTER;
+    entities[0] = character;
+    entities[1] = CreateEntity(ren, 200, 200, 64,128,path);
+    entities[2] = CreateEntity(ren,400,50,64,128,path);
```
Comment (vhakulinen) [inline src/game.c]: If we go this path, query the texture for with and height.

**Comment by wbrbr** [comment]
I think that the bounding box should be separated from the texture dimensions. Maybe we don't want the player's weapon to be part of his hitbox, so we could just reduce the hitbox size to exclude the weapon.

File: src/render.h [hunk 4]
```diff
@@ -15,7 +16,7 @@ SDL_Texture *collision_buffer;
 int buffers_init(SDL_Renderer *ren);
 
 void renderClear(SDL_Renderer*);
-void render(SDL_Renderer*, SDL_Texture*, SDL_Rect*, int color[3]);
+void render(SDL_Renderer*, entity*, int color[3]);
```
Comment (vhakulinen) [inline src/render.h]: I think we'll need to render other stuff to the screen than just entities.

**Comment by vhakulinen** [comment]
Good point

**Comment by wbrbr** [comment]
You're right, I'll change this

**Comment by vhakulinen** [comment]
:+1: looks good for me

**Pull request merged**